omg/entities/projectile.py

```diff
@@ -133,57 +76,24 @@ def create(cls, init_px, init_py, angle) -> Projectile:
         )
 
 
-class FireIceFactory(ProjectileFactory):
-    """Convenience class to create IceShard Projectile."""
-
-    image_file = os.path.join(ASSET_DIR, "skills", "FireIce.PNG")
-    scale = 0.05
-    damage = 15
-    speed = 7
-    mana_cost = 20
-
-    @classmethod
-    def create(cls, init_px, init_py, angle) -> Projectile:
-        """Create an IceShard projectile."""
-        return Projectile(
-            name=cls.__name__,
-            image_file=cls.image_file,
-            scale=cls.scale,
-            damage=cls.damage,
-            init_px=init_px,
-            init_py=init_py,
-            speed=cls.speed,
-            angle=angle,
-        )
-
+with open(CRAFTED_SKILLS_JSON_DIR, "r") as file:
+    crafted_skill_dictionary: Dict[str, Dict] = json.load(file)
 
-class FireFireFactory(ProjectileFactory):
-    """Convenience class to create IceShard Projectile."""
 
-    image_file = os.path.join(ASSET_DIR, "skills", "FireFire.PNG")
-    scale = 0.05
-    damage = 15
-    speed = 7
-    mana_cost = 20
+class CraftingSkillFactory(ProjectileFactory):
+    """Class to combine elements to craft a skill."""
 
     @classmethod
-    def create(cls, init_px, init_py, angle) -> Projectile:
-        """Create an IceShard projectile."""
-        return Projectile(
-            name=cls.__name__,
-            image_file=cls.image_file,
-            scale=cls.scale,
-            damage=cls.damage,
-            init_px=init_px,
-            init_py=init_py,
-            speed=cls.speed,
-            angle=angle,
-        )
-
-
-COMBINED_ELEMENT_DICTIONARY = {
-    "FireFire": FireFireFactory(),
-    "FireIce": FireIceFactory(),
-    "IceFire": IceFireFactory(),
-    "IceIce": IceIceFactory(),
-}
+    def _set_skill_attributes(cls, skill_name: str):
```

Comment:
I don't think class attributes are not the way to go anymore. 

You changed the logic about creating the projectiles. Previously, each class represented a different skill ( now called an element), then you can store the values as a class property as it will be `global` across the lifetime of the game. But now, you only have a single class `CraftingSkillFactory` and all skills are created using this class.

Now I think it makes more sense to create the projectiles based on `instance` attributes, rather than the `class` attributes. Otherwise it would be risky if more than a single user try to create skills, whose attributes will be read from the same **shared** location (which is the class attribute).

You already use an instance of this class in player.py, so you can just refactor the logic easily.